packages/runtime/src/helpers/functions.ts

```diff
@@ -47,19 +58,16 @@ export const generateFunctions = async (
     ? relative(functionDir, nextServerModuleAbsoluteLocation)
     : undefined
 
-  for (const { route, config, compiled } of apiRoutes) {
-    // Don't write a lambda if the runtime is edge
-    if (isEdgeConfig(config.runtime)) {
-      continue
-    }
-    const apiHandlerSource = await getApiHandler({
-      page: route,
-      config,
+  for (const apiLambda of apiLambdas) {
+    const { functionName, routes, type, includedFiles } = apiLambda
+
+    const apiHandlerSource = getApiHandler({
+      schedule: type === ApiRouteType.SCHEDULED ? routes[0].config.schedule : undefined,
```

Comment:
`routes[0]` seems like a code-smell or at least it's not entirely clear to me why this would be fine. Is the data structure of `routes` the problem here, is it fine to only check the first entry or shouldn't we check every route?